### PR TITLE
Adding hiCentrality and hiClusterCompatibility for XeXe re-reco

### PR DESCRIPTION
**Pull request by tuos**
Produce and keep the hiCentrality and hiClusterCompatibility objects
To be used for XeXe re-reco

### CODE REVIEW

File: RecoEgamma/Configuration/python/RecoEgamma_cff.py [hunk 1]
```diff
@@ -48,10 +48,19 @@
 from RecoHI.HiEgammaAlgos.photonIsolationHIProducer_cfi import photonIsolationHIProducerpp
 from RecoHI.HiEgammaAlgos.photonIsolationHIProducer_cfi import photonIsolationHIProducerppGED
 from RecoHI.HiEgammaAlgos.photonIsolationHIProducer_cfi import photonIsolationHIProducerppIsland
+from RecoHI.HiCentralityAlgos.HiCentrality_cfi import hiCentrality
+from RecoHI.HiCentralityAlgos.HiClusterCompatibility_cfi import hiClusterCompatibility
+pp_on_XeXe_2017.toModify(hiCentrality,
```
Comment (perrotta) [inline RecoEgamma/Configuration/python/RecoEgamma_cff.py]: All changes to a module should be done in the same python file that the module is declared in [1]. Please move it inside RecoHI.HiCentralityAlgos.HiCentrality_cfi

[1] https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideCmsDriverEras#Configuring_using_an_era

File: RecoEgamma/Configuration/python/RecoEgamma_cff.py [hunk 2]
```diff
@@ -48,10 +48,19 @@
...
+                         producePixelTracks = False,
+                         srcTracks = cms.InputTag("generalTracks"),
+                         srcVertex = cms.InputTag("offlinePrimaryVertices")
+                         )
 
 _egammaHighLevelRecoPostPF_HI = egammaHighLevelRecoPostPF.copy()
 _egammaHighLevelRecoPostPF_HI += photonIsolationHIProducerpp
 _egammaHighLevelRecoPostPF_HI += photonIsolationHIProducerppGED
 _egammaHighLevelRecoPostPF_HI += photonIsolationHIProducerppIsland
+_egammaHighLevelRecoPostPF_HI += hiCentrality
```
Comment (perrotta) [inline RecoEgamma/Configuration/python/RecoEgamma_cff.py]: I wonder if egammaHighLevelRecoPostPF is the most appropriate place where to include hiCentrality and hiClusterCompatibility? Shouldn't they be more conveniently kept separated inside the highlevelreco sequence of /Configuration/StandardSequences/python/Reconstruction_cff.py

File: RecoEgamma/Configuration/python/RecoEgamma_cff.py (hunk 1 above)
Comment (tuos) [reply RecoEgamma/Configuration/python/RecoEgamma_cff.py]: Done in the new commit

File: RecoEgamma/Configuration/python/RecoEgamma_cff.py (hunk 2 above)
Comment (tuos) [reply RecoEgamma/Configuration/python/RecoEgamma_cff.py]: Done in the new commit

**Comment by tuos** [comment]
@mandrenguyen I did another commit following @perrott 's suggestions. All the previous changes to RecoEgamma/Configuration are removed.  RecoHI/HiCentralityAlgos is updated to make the changes to the module.  Configuration/StandardSequences/python/Reconstruction_cff.py is updated to include the reco sequence following @perrott 's suggestions. The event content is kept in "RecoJets/Configuration/python/RecoJets_EventContent_cff.py" since 'keep *_towerMaker_*_*' was also done in the cff.

**Comment by davidlange6** [comment]
Your PR is unmergeable. Please have a look and possibly rebase it.

**Comment by perrotta** [comment]
Still having conflicts in Configuration/StandardSequences/python/Reconstruction_cff.py

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
+1
- hiCentrality and hiClusterCompatibility added in the reco sequence and Reco/AOD event contents for XeXe reprocessing
- For the reprocessing, a backport is also likely needed

**Comment by tuos** [comment]
Thanks @perrotta a lot for the help. @mandrenguyen said before that he will take care of the backport for reprocessing.

**Comment by tuos** [comment]
Backport is started with #21277

**Comment by davidlange6** [comment]
merge

**Pull request merged**